---
title: "About us"
description: "The Pathogen Informatics and Modelling group"
featured_image: '/images/header2.jpg'
type: 'page'
menu:
  main:
    weight: 1
---

We are the [Pathogen Informatics and Modelling](https://www.ebi.ac.uk/research/lees/) group
at EMBL-EBI (also known as the Lees Group). On this page you can find information about
current and past group members, who we work with, and some nice pictures.

{{< toc >}}

## Lab members

**Hsueh-Chien (Raymond) Cheng** is a visiting PhD student (based at the Wellcome Sanger Institute)

{{< figure src="/images/group/raymond_cheng.jpg" >}}

I am a PhD student at the Wellcome Sanger Institute in [Prof. Stephen Bentley’s group](https://bentleygroup.sanger.ac.uk/#team) and also under the supervision of Dr. John Lees at the EBI. My current work is focusing on quantifying the invasiveness of *Streptococcus pneumoniae* with different genetic backgrounds by using the WGS data among populations. I am also interested in studying various modes of evolution in pathogenic bacteria and how they shape different population structures.

*Languages*: Python, R

_raymond at ebi dot ac dot uk_

**Joel Hellewell** is an [EIPOD](https://www.embl.org/about/info/postdoctoral-programme/eipod-linc-exploring-life-in-context/) fellow

{{< figure src="/images/group/joel_hellewell.jpeg" >}}

My background is in transmission modelling and statistics. I did my PhD at Imperial College London in the Malaria group with Tom Churcher and Azra Ghani. I then did a post-doc at the London School of Hygiene and Tropical Medicine in Sebastian Funk's group. I contributed towards models that performed routine short-term forecasts of COVID-19 deaths, hospitalisations, and reported cases in the United Kingdom. I also worked on reproduction number estimates and modelling the sensitivity of PCR tests over the course of a COVID-19 infection.

I am now interested in integrating genetic and epidemiological models and data to see if this improves the answers to key questions that were repeatedly asking during the COVID-19 pandemic. For example, how early can we tell that the number of infections is increasing? Does this novel variant have a transmission advantage? Does this novel variant cause infections with more serious outcomes?

*Languages and tools*: R, stan

_joel at ebi dot ac dot uk_

{{< icon_wrapper name="Joel" >}}
    {{< social_icon github "https://github.com/jhellewell14" >}}
    {{< social_icon twitter "https://mobile.twitter.com/HellewellJoel" >}}
{{< /icon_wrapper >}}

**Samuel Horsfield** is a postdoctoral fellow

{{< figure src="/images/group/sam_horsfield_photo.jpg" >}}

My PhD focused on pangenome evolution in pathogenic bacteria. I have designed and
developed [ggcaller](https://github.com/samhorsfield96/ggCaller), a new annotation tool
for bacterial pangenomes that works on genome graphs. Other projects include using
graph methods to improve adaptive sequencing of the accessory genome, and using
large population datasets to compare rates of core and accessory genome evolution.

*Languages and tools*: C++, python

*Organisms*: Bacteria (*S. pneumoniae*, *E. coli* and many more) and viruses (polio).

_shorsfield at ebi dot ac dot uk_

{{< icon_wrapper name="Sam" >}}
    {{< social_icon github "https://github.com/samhorsfield96" >}}
    {{< social_icon twitter "https://mobile.twitter.com/samuelhorsfield" >}}
    {{< social_icon linkedin "https://uk.linkedin.com/in/samuel-horsfield-280754101" >}}
{{< /icon_wrapper >}}

**John Lees** is a research group leader

{{< figure src="/images/group/john_lees_photo.jpg" >}}

My previous work has attempted to understand the pathogenesis and transmission
of infectious diseases. I primarily works on statistical genetics, genomic epidemiology and
genome evolution of pathogenic bacteria. I have also worked on infectious disease
modelling, GPU algorithms, and visualisation. I write methods and software to scale
these analyses and make them more accessible to others in attempts to democratise bioinformatics.

*Languages and tools*: C++, CUDA, python, R, rust, stan

*Organisms*: Any microbe, but especially *Streptococcal* bacteria.

My personal website is www.johnlees.me and you can find [my papers here](https://scholar.google.co.uk/citations?user=UVV3P4UAAAAJ&hl=en).

_jlees at ebi dot ac dot uk_

{{< icon_wrapper name="John" >}}
    {{< social_icon github "https://github.com/johnlees" >}}
    {{< social_icon twitter "https://mobile.twitter.com/johnlees6" >}}
{{< /icon_wrapper >}}

**Leonie Lorenz** is a PhD student

{{< figure src="/images/group/leonie_lorenz.jpg" >}}

In my PhD, I will focus on jointly using epidemiological and genomic data for developing mathematical models of pathogen transmission. While the spreading and evolution of pathogens are
often analysed separately, exploiting both data streams together as input for epidemiological models could increase the models' predictive power and result in new insights.

I studied biomathematics at the University of Greifswald, Germany. For my master thesis, I developed a Boolean model for the regulation of apoptosis at the Max Delbrück Center for Molecular Medicine in
the Helmholtz Association.

*Languages and tools*: python, R

_llorenz at ebi dot ac dot uk_

{{< icon_wrapper name="Leonie" >}}
    {{< social_icon github "https://github.com/LeonieLorenz" >}}
    {{< social_icon linkedin "https://www.linkedin.com/in/leonie-lorenz-8629a2a3/" >}}
{{< /icon_wrapper >}}

**Tim Russell** is a visiting postdoc

{{< figure src="/images/group/tim_russell.jpg" >}}

My background is primarily in mathematical biology. I did my PhD at Royal
Holloway University, supervised by [Vincent
Jansen](https://scholar.google.com/citations?user=SrGI1oMAAAAJ&hl=en&oi=ao),
studying mathematical aspects of models from theoretical population genetics. I
then started a postdoc at London School of Hygiene and Tropical Medicine in
Adam Kucharski's group, originally to work on modelling antibody dynamics for
influenza. The COVID-19 pandemic began shortly after I started (!), so I ended
up working intensely on several aspects of the pandemic. These included:
transmission dynamics, severity estimates, global estimates of infections, the
sensitivity of PCR and LFT tests, within-host viral kinetics, within-host
antibody dynamics, etc. We have have recently started a long-term project in
collaboration with The Francis Crick Institute, setting up a data generating
and analysis pipeline, with a large modelling component, studying how antibody
levels for COVID-19 vary between different regions and over time.

I aim to use my time at EBI learning more about genomics. Specifically,
thinking about whether it is possible to combine models of within-host
responses to infection with genomic data data to arrive at detailed estimates
of how hosts and pathogens coevolve.

*Languages and tools*: C++, R, Stan, bash

*Organisms*: SARS-CoV-2, Influenza, MPX

_thimotei at ebi dot ac dot uk_

{{< icon_wrapper name="Tim" >}}
    {{< social_icon github "https://github.com/thimotei" >}}
    {{< social_icon twitter "https://twitter.com/timwrussell" >}}
{{< /icon_wrapper >}}

**Jacqueline Toussaint** is a PhD student

{{< figure src="/images/group/jacqueline_toussaint.jpg" >}}

I'm broadly focused on developing more efficient and accurate methods for performing genome-wide association studies (GWAS) in bacteria, with a particular focus on antibiotic resistance and the possibility of expanding to real-time analysis. I'm also working on developing methods to improve the speed and consistency of bacterial genome annotation.

I previously studied biochemistry and microbiology at Montana State University, USA, with my Bachelor's thesis research focused on generating and analyzing multi-omics datasets. I’m excited to further my education in the lab of Dr. John Lees at EMBL-EBI and the University of Cambridge, UK.

*Languages and tools*: python, R, Rust, Stan

*Organisms:* I work with a diverse set of microbes, including *Streptococcus*, *Klebsiella*, and *Neisseria*.

_lilyjacqueline at ebi dot ac dot uk_

{{< icon_wrapper name="Jacqueline" >}}
    {{< social_icon github "https://github.com/qtoussaint" >}}
    {{< social_icon linkedin "https://uk.linkedin.com/in/jackie-toussaint" >}}
{{< /icon_wrapper >}}

**Johanna von Wachsmann** is a PhD student (based in [Rob Finn's](https://www.ebi.ac.uk/research/finn/) lab).

{{< figure src="/images/group/johanna_von_wachsmann_photo.jpeg" >}}

I did my master’s degree in bioinformatics with a focus on advanced algorithms at the Freie Universität Berlin. I wrote my master’s thesis under the supervision of Dr. John Lees.

I am now working on developing tools and algorithms for the analysis of large metagenomic
datasets.

*Languages*: C++, Python, R

_wachsmannj at ebi dot ac dot uk_

{{< icon_wrapper name="Johanna" >}}
    {{< social_icon github "https://github.com/johannahelene" >}}
    {{< social_icon twitter "https://twitter.com/vWachsmann" >}}
    {{< social_icon linkedin "https://www.linkedin.com/in/johanna-helene-von-wachsmann-191b91224/" >}}
{{< /icon_wrapper >}}


### Alumni

**Gherard Batisti Biffignandi** (2022) was a visiting PhD student (based at the University of Pavia). Now at the University of Cambridge (Bakery lab).

**Bruhad Dave** (2022) was an intern, now at the Wellcome Sanger Institute.

**Berk Gonenc** (2023) was an intern.

**Tommi Mäklin** (2023) was a visiting postdoc on the Theory@EMBL visitor scheme (based at the University of Helsinki).

**Oladipo Omotosho** (2023) was a visiting scientist on an EMBL training fellowship (based at the University of Ibidan).

**Tendai Washaya** (2023) was a visiting PhD student (based at the University of Zimbabwe)

## Social and photos

### 2023

### Xmas dinner 2023

{{< figure src="/images/group/2023_xmas_1.jpg" >}}

{{< figure src="/images/group/2023_xmas_2.jpg" >}}

#### Group retreat, Suffolk. July 2023

{{< figure src="/images/group/2023_retreat_3.jpeg" >}}

{{< figure src="/images/group/2023_retreat_1.jpeg" >}}

{{< figure src="/images/group/2023_retreat_2.jpeg" >}}

{{< figure src="/images/group/2023_retreat_4.jpeg" >}}

{{< figure src="/images/group/2023_retreat_5.jpeg" caption="The finest minds" >}}

{{< figure src="/images/group/2023_retreat_6.jpeg" caption="The finest minds (part 2)">}}

{{< figure src="/images/group/2023_retreat_7.jpeg" >}}

#### Pub bike May 2023 -- the hailstorm

{{< figure src="/images/group/pubbike1.jpeg" >}}

{{< figure src="/images/group/pubbike2.jpeg" >}}

{{< figure src="/images/group/pubbike3.jpeg" >}}

{{< figure src="/images/group/pubbike4.jpeg" >}}


### 2022

{{< figure src="/images/group/2022_ib_tt.jpg" caption="Infection Biology Transversal Theme Retreat">}}

{{< figure src="/images/group/2022_zoo_1.jpg" caption="Group retreat at London Zoo (Gorilla sadly not in our group)">}}

{{< figure src="/images/group/2022_zoo_2.jpg" >}}
