---
title: "About us"
description: "The Pathogen Informatics and Modelling group"
featured_image: '/images/header2.jpg'
type: 'page'
menu:
  main:
    weight: 1
---

We are the [Pathogen Informatics and Modelling](https://www.ebi.ac.uk/research/lees/) group
at EMBL-EBI (also known as the Lees Group). On this page you can find information about
current and past group members, who we work with, and some nice pictures.

{{< toc >}}

## Lab members

**Samuel Horsfield** is a postdoctoral fellow

{{< figure src="/images/group/sam_horsfield_photo.jpg" >}}

My PhD focused on pangenome evolution in pathogenic bacteria. I designed and
developed [ggcaller](https://github.com/samhorsfield96/ggCaller), a new annotation tool
for bacterial pangenomes that works on pangenome graphs. In my Postdoc, I'm working on
graph methods to improve Nanopore sequencing for enrichment of the accessory genome, using
large population datasets to understand pangenome evolution, and developing large language models to simulate bacterial populations.

*Languages and tools*: C++, Rython, R

*Organisms*: Bacteria (*S. pneumoniae*, *E. coli* and many more) and viruses (polio).

_shorsfield at ebi dot ac dot uk_

{{< icon_wrapper name="Sam" >}}
    {{< social_icon github "https://github.com/samhorsfield96" >}}
    {{< social_icon twitter "https://mobile.twitter.com/samuelhorsfield" >}}
    {{< social_icon linkedin "https://uk.linkedin.com/in/samuel-horsfield-280754101" >}}
{{< /icon_wrapper >}}

**John Lees** is a research group leader

{{< figure src="/images/group/john_lees_photo.jpg" >}}

My previous work has attempted to understand the pathogenesis and transmission
of infectious diseases. I primarily work on statistical genetics, genomic epidemiology and
genome evolution of pathogenic bacteria. I have also worked on infectious disease
modelling, GPU algorithms, and visualisation. I write methods and software to scale
these analyses and make them more accessible to others in attempts to democratise bioinformatics.

*Languages and tools*: C++, CUDA, python, R, rust, stan

*Organisms*: Any microbe, but especially *Streptococcal* bacteria.

My personal website is www.johnlees.me and you can find [my papers here](https://scholar.google.co.uk/citations?user=UVV3P4UAAAAJ&hl=en).

_jlees at ebi dot ac dot uk_

{{< icon_wrapper name="John" >}}
    {{< social_icon github "https://github.com/johnlees" >}}
    {{< social_icon twitter "https://mobile.twitter.com/johnlees6" >}}
{{< /icon_wrapper >}}

**Leonie Lorenz** is a PhD student

{{< figure src="/images/group/leonie_lorenz.jpg" >}}

In my PhD, I will focus on jointly using epidemiological and genomic data for developing mathematical models of pathogen transmission. While the spreading and evolution of pathogens are
often analysed separately, exploiting both data streams together as input for epidemiological models could increase the models' predictive power and result in new insights.

I studied biomathematics at the University of Greifswald, Germany. For my master thesis, I developed a Boolean model for the regulation of apoptosis at the Max Delbrück Center for Molecular Medicine in
the Helmholtz Association.

*Languages and tools*: python, R

_llorenz at ebi dot ac dot uk_

{{< icon_wrapper name="Leonie" >}}
    {{< social_icon github "https://github.com/LeonieLorenz" >}}
    {{< social_icon linkedin "https://www.linkedin.com/in/leonie-lorenz-8629a2a3/" >}}
{{< /icon_wrapper >}}

**Neil MacAlasdair** is an [EIPOD](https://www.embl.org/about/info/postdoctoral-programme/eipod-linc-exploring-life-in-context/) fellow

{{< figure src="/images/group/neil_macalasdair.jpg" >}}

My fellowship project studies speciation in microbes -- particularly in two genera of upper respiratory tract bacteria, _Neisseria_ and _Haemophilus_ -- using large collections of whole-genome sequence data. These are opportune genera for this research because each contains well-sampled disease causing species, they inhabit the same niche (the human nasopharynx) and have similar life histories, but are distantly related. To advance this work, I use and develop pangenomic methods to cope with the scale and diversity of genus-wide data. 

Previously, I was a [Scientia Fellow](https://www.med.uio.no/english/research/scientia-fellows/our-fellows/population-genomics-gene-regulation/) with [Jukka Corander](https://www.med.uio.no/imb/english/research/groups/probabilistic-inference-lab/index.html) at the University of Oslo, where I primarily studied the global population structure of _Haemophilus influenzae_ and developed methods to infer recombination-free phylogenies from core genome alignments derived from pangenome methods such as panaroo. Prior to that I was a [PhD student](https://www.sanger.ac.uk/person/macalasdair-neil/) at the Wellcome Sanger Institute with [Stephen Bentley](https://www.sanger.ac.uk/group/genomics-of-pneumonia-meningitis-and-neonatal-sepsis/), during which I studied the global population strucure and pangenome of _Neisseria meningitidis_.

_macalasdair at ebi dot ac dot uk_

{{< icon_wrapper name="Neil" >}}
    {{< social_icon github "https://github.com/nzmacalasdair" >}}
{{< /icon_wrapper >}}

**Anlei Peng** is an intern

{{< figure src="/images/group/anlei_peng_photo.jpeg" >}}

I am working on automating quality control and model fitting in PopPUNK for clustering large genome datasets at the sub-species level.

I did my Bachelor's degree and integrated master's degree at the University of Cambridge with a focus on Genetics and Systems Biology.

*Languages*: Python, R

*Organisms:* I worked on *Orientia tsutsugamushi* in my master's project in Dr Jeanne Salje's group at CIMR.

_apeng at ebi dot ac dot uk_

{{< icon_wrapper name="Anlei" >}}
    {{< social_icon github "https://github.com/anleip" >}}
    {{< social_icon linkedin "https://www.linkedin.com/in/anlei-peng-13005622a/" >}}
{{< /icon_wrapper >}}

**Víctor Rodríguez Bouza** is a Postdoctoral Fellow.

{{< figure src="/images/group/victor_rodriguezbouza.jpeg" >}}

After high school, I started a BSc in Bioinformatics, but I decided to change after a year to Mathematics and Physics, both in the Universidad de Oviedo (Asturias, Spain). After a MSc in High Energy Physics (in the UAB in Barcelona, Spain), I did my PhD thesis in the same field in the Universidad de Oviedo, as a member of the CMS Collaboration at CERN. Now I have returned to the bio-sciences and started as a postdoctoral fellow in the Lees Group.


*Languages, tools*: Python, C++, R, ROOT.

~~Organisms~~ *Quarks*: I have worked mainly in top quark physics.

_vrbouza at ebi dot ac dot uk_

{{< icon_wrapper name="Víctor" >}}
    {{< social_icon github "https://github.com/vrbouza" >}}
    {{< social_icon linkedin "https://www.linkedin.com/in/vrbouza" >}}
{{< /icon_wrapper >}}

**Matt Russell** is a postdoctoral fellow.

{{< figure src="/images/group/matt_russell.jpg" >}}

Currently, I'm working on a project to catalogue and characterise bacterial capsules, which are important for both infection and as vaccination targets, across all sequenced bacteria. In addition to this, I am improving the GPU backend of the `dust` stochastic simulation package. The idea is that we can exploit the dependency structure of the underlying model to automatically produce a more efficient set of GPU kernels.

My background is in applied mathematics, mathematical biology, and computer science. My PhD was supervised by Oliver Jensen and Tobias Galla at the University of Manchester and focussed on mathematical models of transport processes in spatially disordered media. After that, I was a postdoc at the University of Nottingham working with Bindi Brook on models of chemokine gradient formation and immune cell migration.

*Languages and tools*: C++, CUDA, Mathematica, R, Rust, Stan

_mjrussell at ebi dot ac dot uk_

{{< icon_wrapper name="Matt" >}}
    {{< social_icon github "https://github.com/mtrsl" >}}
{{< /icon_wrapper >}}

**Jacqueline Toussaint** is a PhD student

{{< figure src="/images/group/jacqueline_toussaint.jpg" >}}

I'm broadly focused on developing more efficient and accurate methods for performing genome-wide association studies (GWAS) in bacteria, with a particular focus on antibiotic resistance and the possibility of expanding to real-time analysis. I'm also working on developing methods to improve the speed and consistency of bacterial genome annotation.

I previously studied biochemistry and microbiology at Montana State University, USA, with my Bachelor's thesis research focused on generating and analyzing multi-omics datasets. I’m excited to further my education in the lab of Dr. John Lees at EMBL-EBI and the University of Cambridge, UK.

*Languages and tools*: python, R, Rust, Stan

*Organisms:* I work with a diverse set of microbes, including *Streptococcus*, *Klebsiella*, and *Neisseria*.

_lilyjacqueline at ebi dot ac dot uk_

{{< icon_wrapper name="Jacqueline" >}}
    {{< social_icon github "https://github.com/qtoussaint" >}}
    {{< social_icon linkedin "https://uk.linkedin.com/in/jackie-toussaint" >}}
{{< /icon_wrapper >}}

**Johanna von Wachsmann** is a PhD student (based in [Rob Finn's](https://www.ebi.ac.uk/research/finn/) lab).

{{< figure src="/images/group/johanna_von_wachsmann_photo.jpeg" >}}

I did my master’s degree in bioinformatics with a focus on advanced algorithms at the Freie Universität Berlin. I wrote my master’s thesis under the supervision of Dr. John Lees.

I am now working on developing tools and algorithms for the analysis of large metagenomic
datasets.

*Languages*: C++, Python, R

_wachsmannj at ebi dot ac dot uk_

{{< icon_wrapper name="Johanna" >}}
    {{< social_icon github "https://github.com/johannahelene" >}}
    {{< social_icon twitter "https://twitter.com/vWachsmann" >}}
    {{< social_icon linkedin "https://www.linkedin.com/in/johanna-helene-von-wachsmann-191b91224/" >}}
{{< /icon_wrapper >}}

**Ouli Xie** is an [ESPOD](https://www.ebi.ac.uk/research/postdocs/espods/) fellow

{{< figure src="/images/group/ouli_xie.jpg" >}}

I am an infectious diseases physician-researcher with an interest in applied bacterial genomics to understand how pathogens evolve and spread. I completed my PhD at the Doherty Institute, Australia, with [Steven Tong](https://www.doherty.edu.au/people/professor-steven-tong) and [Mark Davies](https://www.doherty.edu.au/people/dr-mark-davies). In my PhD I examined the clinical and evolutionary intersection between pathogenic streptococci integrating large-scale pathogen genomics, clinical, and surveillance data.

In my fellowship, I work across the Lees Group, Bateman Group and [Harrison Group](https://www.sanger.ac.uk/person/harrison-ewan/) at EMBL-EBI and the Wellcome Sanger Institute to study determinants of _Staphylococcus aureus_ carriage. In particular, I examine co-evolution and competition between _Staphylococcus aureus_ and the nasal microbiome during carriage integrating large-scale pathogen, microbiome, and host data.

*Languages and tools*: Python, R

*Organisms*: _Staphylococcus aureus_, streptococci, microbiome

_oxie at ebi dot ac dot uk_

{{< icon_wrapper name="Ouli" >}}
    {{< social_icon github "https://github.com/oulixie" >}}
{{< /icon_wrapper >}}

### Alumni

**Daniel Anderson** (2024-2025) was a rehomed PhD student, supervised by Zam Iqbal. Now at Basecamp Research.

**Raymond Cheng** (2022-2025) was a visiting PhD student, co-supervised with Stephen Bentley at the Wellcome Sanger Institute. Now in Nick Croucher's group at Imperial College London.

**Joel Hellewell** (2022-2025) was a postdoc, then EIPOD fellow, and founding group member. He's now at MSF and the Jean Golding Institute at the University of Bristol.

**Alireza Tajmirriahi** (2025) was a research intern from Aalto University.

**Aya Ben Taghaline** (2025) was a master's student from Université Paris-Saclay.

**Ines Sanchez-Hombria** (2025) was an intern.

**Andrea Sanchez Serrano** (2025) was a visiting PhD student from the University of Valencia.

**Oluwaremilekun Grace Ajakaye** (2024) was a visiting scientist (based at Adekunle Ajasin University).

**Lale Maouloud** (2024) was a master's student from the University of Rouen Normandy.

**Antoine Andréoletti** (2024) was a master's student from the Ecole Polytechnique in Paris.

**Berk Gonenc** (2023) was an intern.

**Oladipo Omotosho** (2023) was a visiting scientist on an EMBL training fellowship (based at the University of Ibidan).

**Tendai Washaya** (2023) was a visiting PhD student (based at the University of Zimbabwe).

**Tommi Mäklin** (2023) was a visiting postdoc on the Theory@EMBL visitor scheme (based at the University of Helsinki).

**Timothy Russell** (2023) was a visiting scientist (based at London School of Hygiene and Tropical Medicine).

**Gherard Batisti Biffignandi** (2022) was a visiting PhD student (based at the University of Pavia). Now at the University of Cambridge ([Kate Baker's lab](https://baker-lab.github.io/)) and still a visitor at PIM.

**Bruhad Dave** (2022) was an intern, now at the Wellcome Sanger Institute.

## Social and photos

### 2025

#### Group retreat, Center Parcs. October 2025

{{< figure src="/images/group/2025_cp_1.jpg">}}

{{< figure src="/images/group/2025_cp_2.jpg">}}

{{< figure src="/images/group/2025_cp_3.jpg">}}

{{< figure src="/images/group/2025_cp_4.jpg"> caption="Collage of destruction"}}


#### EMBO course: Genomics and Modelling. Sept 2025

{{< figure src="/images/group/2025_embo.jpg">}}

#### Group away day, Cambridge/Bury St Edmunds. February 2025

{{< figure src="/images/group/gokarting_1.jpg">}}

{{< figure src="/images/group/gokarting_2.jpg">}}

{{< figure src="/images/group/gokarting_3.jpg" caption="Gracious winners">}}

### 2024

#### Winter 2024

{{< figure src="/images/group/2024_xmas.jpg">}}

{{< figure src="/images/group/pub_engage_1.jpg" caption="Brought to you by Chouffre">}}

#### Public engagement. Oct 2024

{{< figure src="/images/group/pub_engage_2.jpg" >}}

{{< figure src="/images/group/pub_engage_3.jpg" >}}

#### Group retreat, Colchester. August 2024

{{< figure src="/images/group/2024_retreat2_1.jpg" >}}

{{< figure src="/images/group/2024_retreat2_2.jpg" >}}

{{< figure src="/images/group/2024_retreat2_3.jpg" caption="Cultural exchange dinner (jackets)">}}

{{< figure src="/images/group/2024_retreat2_4.jpg" caption="Minigolf with Augustus I">}}

{{< figure src="/images/group/2024_retreat2_5.jpg" caption="pints ~ years as postdoc">}}

{{< figure src="/images/group/2024_retreat2_6.jpg" >}}

{{< figure src="/images/group/2024_retreat2_7.jpg" caption="Temporary group member Hargreaves the cat">}}


#### Microbiology society conference

{{< figure src="/images/group/2024_microsoc_gnasty.JPG" caption="Sam presenting his work on GNASTy" >}}

{{< figure src="/images/group/2024_microsoc_dance.jpg" caption="Sam after presenting his work on GNASTy">}}

#### Group away day, Cambridge. April 2024

{{< figure src="/images/group/2024_retreat_1.jpeg" >}}

{{< figure src="/images/group/2024_retreat_2.jpeg" caption="Lane 1 (high achievers)" >}}

{{< figure src="/images/group/2024_retreat_3.jpeg" caption="Lane 2 (remedial bowling class)">}}

### 2023

#### Xmas dinner 2023

{{< figure src="/images/group/2023_xmas_1.jpg" >}}

{{< figure src="/images/group/2023_xmas_2.jpg" >}}

#### Group retreat, Suffolk. July 2023

{{< figure src="/images/group/2023_retreat_3.jpeg" >}}

{{< figure src="/images/group/2023_retreat_1.jpeg" >}}

{{< figure src="/images/group/2023_retreat_2.jpeg" >}}

{{< figure src="/images/group/2023_retreat_4.jpeg" >}}

{{< figure src="/images/group/2023_retreat_5.jpeg" caption="The finest minds" >}}

{{< figure src="/images/group/2023_retreat_6.jpeg" caption="The finest minds (part 2)">}}

{{< figure src="/images/group/2023_retreat_7.jpeg" >}}

#### Pub bike May 2023 -- the hailstorm

{{< figure src="/images/group/pubbike1.jpeg" >}}

{{< figure src="/images/group/pubbike2.jpeg" >}}

{{< figure src="/images/group/pubbike3.jpeg" >}}

{{< figure src="/images/group/pubbike4.jpeg" >}}


### 2022

{{< figure src="/images/group/2022_ib_tt.jpg" caption="Infection Biology Transversal Theme Retreat">}}

{{< figure src="/images/group/2022_zoo_1.jpg" caption="Group retreat at London Zoo (Gorilla sadly not in our group)">}}

{{< figure src="/images/group/2022_zoo_2.jpg" >}}
