---
title: "About us"
description: "The Pathogen Informatics and Modelling group"
featured_image: '/images/header2.jpg'
type: 'page'
menu:
  main:
    weight: 1
---

We are the [Pathogen Informatics and Modelling](https://www.ebi.ac.uk/research/lees/) group
at EMBL-EBI (also known as the Lees Group). On this page you can find information about
current and past group members, who we work with, and some nice pictures.

{{< toc >}}

## Lab members

**Raymond Cheng** is a visiting PhD student (based at the Wellcome Sanger Institute)

_raymond at ebi dot ac dot uk_

**Bruhad Dave** is an intern

_bruhad at ebi dot ac dot uk_

**Joel Hellewell** is a postdoctoral fellow

_joel at ebi dot ac dot uk_

{{< icon_wrapper name="Joel" >}}
    {{< social_icon github "https://github.com/jhellewell14" >}}
    {{< social_icon twitter "https://mobile.twitter.com/HellewellJoel" >}}
{{< /icon_wrapper >}}

**Samuel Horsfield** is a visiting PhD student (based at Imperial College London)

{{< figure src="/images/group/sam_horsfield_photo.jpg" >}}

My PhD focuses on pangenome evolution in pathogenic bacteria. I have designed and
developed [ggcaller](https://github.com/samhorsfield96/ggCaller), a new annotation tool
for bacterial pangenomes that works on genome graphs. Other projects include using
graph methods to improve adaptive sequencing of the accessory genome, and using
large population datasets to compare rates of core and accessory genome evolution.

*Languages and tools*: C++, python

*Organisms*: Bacteria (*S. pneumoniae*, *E. coli* and many more) and viruses (polio).

_shorsfield at ebi dot ac dot uk_

{{< icon_wrapper name="Sam" >}}
    {{< social_icon github "https://github.com/samhorsfield96" >}}
    {{< social_icon twitter "https://mobile.twitter.com/samuelhorsfield" >}}
    {{< social_icon linkedin "https://uk.linkedin.com/in/samuel-horsfield-280754101" >}}
{{< /icon_wrapper >}}

**John Lees** is a research group leader

{{< figure src="/images/group/john_lees_photo.jpg" >}}

My previous work has attempted to understand the pathogenesis and transmission
of infectious diseases. I primarily works on statistical genetics, genomic epidemiology and
genome evolution of pathogenic bacteria. I have also worked on infectious disease
modelling, GPU algorithms, and visualisation. I write methods and software to scale
these analyses and make them more accessible to others in attempts to democratise bioinformatics.

*Languages and tools*: C++, CUDA, python, R, stan

*Organisms*: Any microbe, but especially *Streptococcal* bacteria.

My personal website is www.johnlees.me and you can find [my papers here](https://scholar.google.co.uk/citations?user=UVV3P4UAAAAJ&hl=en).

_jlees at ebi dot ac dot uk_

{{< icon_wrapper name="John" >}}
    {{< social_icon github "https://github.com/johnlees" >}}
    {{< social_icon twitter "https://mobile.twitter.com/johnlees6" >}}
{{< /icon_wrapper >}}

**Leonie Lorenz** is a PhD student

_leonie at ebi dot ac dot uk_

**Jacqueline Toussaint** is a PhD student

_lilyjacqueline at ebi dot ac dot uk_

**Johanna von Wachsmann** is a Master's student

{{< figure src="/images/group/johanna_von_wachsmann_photo.jpeg" >}}

I am currently earning my master’s degree in bioinformatics with a focus on advanced algorithms at the Freie Universität Berlin. I am writing my master’s thesis under the supervision of Dr. John Lees.
I studied bioinformatics for my master’s and bachelor’s degree. My master’s thesis project involves building an analytical tool [SKA2: split k-mer analysis](https://github.com/bacpop/SKA2), for use in such practical applications as pathogen outbreak analyses. I am developing a Python/C++ package which uses sequence data to produce rapid, multiple-sequence alignments. The storing of k-mers as bit vectors aims to improve run-time and disk usage. I also expanded the genome clustering tool PopPUNK for use in investigation of pathogen outbreaks.

*Languages*: C++, Python, R

_wachsmannj at ebi dot ac dot uk_


### Alumni

**Gherard Batisti Biffignandi** was a visiting PhD student (based at the University of Pavia).
## Social and photos

### 2022 group retreat

{{< figure src="/images/group/2022_zoo_1.jpg" >}}

{{< figure src="/images/group/2022_zoo_2.jpg" >}}
## Collaborators

Outside of EMBL, we currently work with following people and groups on some of our projects:

* Nick Croucher and the [Bacterial Evolutionary Epidemiology Group](https://www.imperial.ac.uk/mrc-global-infectious-disease-analysis/hosted-initiatives-and-groups/bacterial-evolutionary-epidemiology-group/) (Imperial College London).
* Rich Fitzjohn and the [RESIDE Group](https://reside-ic.github.io/) (Imperial College London).
* Stephen Bentley, [his team](https://bentleygroup.sanger.ac.uk/), and the [GPS](https://www.pneumogen.net/gps/) and [Juno](https://www.gbsgen.net/) projects (Wellcome Sanger Institute).
* Jukka Corander and the [Probabilistic Inference Lab](https://www.med.uio.no/imb/english/research/groups/probabilistic-inference-lab/index.html) (University of Oslo).

We're always interested in growing this network, and have short- and medium-term
visitors to the lab. Please get in touch if you'd like
to collaborate with us.

## Working with us

For **masters projects**, please send us an email with your CV including any existing research, software and programming
experience, and your scientific interests.

For **internships**, we normally advertise these in January each year. This may be later in 2023.

For **PhDs**, application is through the [EMBL programme](https://www.embl.org/about/info/embl-international-phd-programme/overview/). We don't currently expect to have a vacancy in the next round.

For **postdocs**, we don't currently have any direct vacancies. However, we are participating
in the current [EIPOD call](https://www.embl.org/about/info/postdoctoral-programme/eipod-linc-exploring-life-in-context/) which runs from October 17th 2022 until January 19th 2023.
Please contact us if you have any idea for a project you'd like to work on.

We don't currently have any vacancies for **software developers**.
