---
title: "About us"
description: "The Pathogen Informatics and Modelling group"
featured_image: '/images/header2.jpg'
type: 'page'
menu:
  main:
    weight: 1
---

We are the [Pathogen Informatics and Modelling](https://www.ebi.ac.uk/research/lees/) group
at EMBL-EBI (also known as the Lees Group). On this page you can find information about
current and past group members, who we work with, and some nice pictures.

{{< toc >}}

## Lab members

**Raymond Cheng** is a visiting PhD student (based at the Wellcome Sanger Institute)

I am a PhD student at Sanger Institute in [Prof. Stephen Bentley’s group](https://www.sanger.ac.uk/group/genomics-pneumonia-and-meningitis-and-neonatal-sepsis/) and also under the supervision of Dr. John Lees at the EBI. My current work is focusing on quantifying the invasiveness of *Streptococcus pneumoniae* with different genetic backgrounds by using the WGS data among populations. I am also interested in studying various modes of evolution in pathogenic bacteria and how they shape different population structures. 

*Languages*: Python, R

_raymond at ebi dot ac dot uk_

**Bruhad Dave** is an intern

My work in the lab involves writing a [python package](https://github.com/bacpop/pp-netlib) that handles graph data for the population genomics tool, [PopPUNK](https://github.com/bacpop/PopPUNK), and contributing to the analysis of patterns of pneumococcal genomic variation. My past work has included looking at segmental variants in lab-evolved lines of drug-resistant _E. coli_, assembling and annotating data from nosocomial samples, and investigating the extent, rates, and predicted functional effects of single-nucleotide variations in the transcriptome of _Plasmodium falciparum_. I am broadly interested in human health & disease research, and I am especially interested in the intersections of pathogen genomics, host-pathogen interactions, and evolutionary biology. 

*Languages*: Python, R, Bash/Shell

{{< icon_wrapper name="Bruhad" >}}
    {{< social_icon github "https://github.com/bruhad-dave" >}}
{{< /icon_wrapper >}}

_bruhad at ebi dot ac dot uk_

**Joel Hellewell** is a postdoctoral fellow

{{< figure src="/images/group/joel_hellewell.jpeg" >}}

My background is in transmission modelling and statistics. I did my PhD at Imperial College London in the Malaria group with Tom Churcher and Azra Ghani. I then did a post-doc at the London School of Hygiene and Tropical Medicine in Sebastian Funk's group. I contributed towards models that performed routine short-term forecasts of COVID-19 deaths, hospitalisations, and reported cases in the United Kingdom. I also worked on reproduction number estimates and modelling the sensitivity of PCR tests over the course of a COVID-19 infection.

I am now interested in integrating genetic and epidemiological models and data to see if this improves the answers to key questions that were repeatedly asking during the COVID-19 pandemic. For example, how early can we tell that the number of infections is increasing? Does this novel variant have a transmission advantage? Does this novel variant cause infections with more serious outcomes?

*Languages and tools*: R, stan

_joel at ebi dot ac dot uk_

{{< icon_wrapper name="Joel" >}}
    {{< social_icon github "https://github.com/jhellewell14" >}}
    {{< social_icon twitter "https://mobile.twitter.com/HellewellJoel" >}}
{{< /icon_wrapper >}}

**Samuel Horsfield** is a visiting PhD student (based at Imperial College London)

{{< figure src="/images/group/sam_horsfield_photo.jpg" >}}

My PhD focuses on pangenome evolution in pathogenic bacteria. I have designed and
developed [ggcaller](https://github.com/samhorsfield96/ggCaller), a new annotation tool
for bacterial pangenomes that works on genome graphs. Other projects include using
graph methods to improve adaptive sequencing of the accessory genome, and using
large population datasets to compare rates of core and accessory genome evolution.

*Languages and tools*: C++, python

*Organisms*: Bacteria (*S. pneumoniae*, *E. coli* and many more) and viruses (polio).

_shorsfield at ebi dot ac dot uk_

{{< icon_wrapper name="Sam" >}}
    {{< social_icon github "https://github.com/samhorsfield96" >}}
    {{< social_icon twitter "https://mobile.twitter.com/samuelhorsfield" >}}
    {{< social_icon linkedin "https://uk.linkedin.com/in/samuel-horsfield-280754101" >}}
{{< /icon_wrapper >}}

**John Lees** is a research group leader

{{< figure src="/images/group/john_lees_photo.jpg" >}}

My previous work has attempted to understand the pathogenesis and transmission
of infectious diseases. I primarily works on statistical genetics, genomic epidemiology and
genome evolution of pathogenic bacteria. I have also worked on infectious disease
modelling, GPU algorithms, and visualisation. I write methods and software to scale
these analyses and make them more accessible to others in attempts to democratise bioinformatics.

*Languages and tools*: C++, CUDA, python, R, stan

*Organisms*: Any microbe, but especially *Streptococcal* bacteria.

My personal website is www.johnlees.me and you can find [my papers here](https://scholar.google.co.uk/citations?user=UVV3P4UAAAAJ&hl=en).

_jlees at ebi dot ac dot uk_

{{< icon_wrapper name="John" >}}
    {{< social_icon github "https://github.com/johnlees" >}}
    {{< social_icon twitter "https://mobile.twitter.com/johnlees6" >}}
{{< /icon_wrapper >}}

**Leonie Lorenz** is a PhD student

_leonie at ebi dot ac dot uk_

**Jacqueline Toussaint** is a PhD student

_lilyjacqueline at ebi dot ac dot uk_

**Johanna von Wachsmann** is a Master's student

{{< figure src="/images/group/johanna_von_wachsmann_photo.jpeg" >}}

I am currently earning my master’s degree in bioinformatics with a focus on advanced algorithms at the Freie Universität Berlin. I am writing my master’s thesis under the supervision of Dr. John Lees.
I studied bioinformatics for my master’s and bachelor’s degree. My master’s thesis project involves building an analytical tool [SKA2: split k-mer analysis](https://github.com/bacpop/SKA2), for use in such practical applications as pathogen outbreak analyses. I am developing a Python/C++ package which uses sequence data to produce rapid, multiple-sequence alignments. The storing of k-mers as bit vectors aims to improve run-time and disk usage. I also expanded the genome clustering tool PopPUNK for use in investigation of pathogen outbreaks.

*Languages*: C++, Python, R

_wachsmannj at ebi dot ac dot uk_


### Alumni

**Gherard Batisti Biffignandi** was a visiting PhD student (based at the University of Pavia).
## Social and photos

### 2022 group retreat

{{< figure src="/images/group/2022_zoo_1.jpg" >}}

{{< figure src="/images/group/2022_zoo_2.jpg" >}}
