<!DOCTYPE html>
<html lang="en-gb">
  <head><script src="/livereload.js?mindelay=10&amp;v=2&amp;port=1313&amp;path=livereload" data-no-instant defer></script>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    
    <title>Building trees with SKA | Pathogen Informatics and Modelling Group</title>
    <meta name="viewport" content="width=device-width,minimum-scale=1">
    <meta name="description" content="SKA is a tool for comparing small and highly similar genomes using split k-mers. This guide will explain how to use SKA to build a phylogenetic tree for different Escherichia coli lineages in a few minutes. Although SKA is tailored more towards analysing variation within a lineage, tree-building ends up working fine for the whole species but requires more memory.
Why SKA is good for building phylogenetic trees The basic approach to building a tree with SKA is to generate a SNP alignment using split k-mers and then feed that to a tree building algorithm of choice.">
    <meta name="generator" content="Hugo 0.124.1">
    
    
    
    
      <meta name="robots" content="noindex, nofollow">
    

    
<link rel="stylesheet" href="/ananke/css/main.min.css" >



    
    
    
      
<link rel="shortcut icon" href="/images/favicon.svg" type="image/x-icon" />


    

    
    
    <meta property="og:title" content="Building trees with SKA" />
<meta property="og:description" content="SKA is a tool for comparing small and highly similar genomes using split k-mers. This guide will explain how to use SKA to build a phylogenetic tree for different Escherichia coli lineages in a few minutes. Although SKA is tailored more towards analysing variation within a lineage, tree-building ends up working fine for the whole species but requires more memory.
Why SKA is good for building phylogenetic trees The basic approach to building a tree with SKA is to generate a SNP alignment using split k-mers and then feed that to a tree building algorithm of choice." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://localhost:1313/guides/building_trees_with_ska/" /><meta property="article:section" content="guides" />
<meta property="article:published_time" content="2023-07-21T16:22:28+01:00" />
<meta property="article:modified_time" content="2023-07-21T16:22:28+01:00" />
<meta itemprop="name" content="Building trees with SKA">
<meta itemprop="description" content="SKA is a tool for comparing small and highly similar genomes using split k-mers. This guide will explain how to use SKA to build a phylogenetic tree for different Escherichia coli lineages in a few minutes. Although SKA is tailored more towards analysing variation within a lineage, tree-building ends up working fine for the whole species but requires more memory.
Why SKA is good for building phylogenetic trees The basic approach to building a tree with SKA is to generate a SNP alignment using split k-mers and then feed that to a tree building algorithm of choice."><meta itemprop="datePublished" content="2023-07-21T16:22:28+01:00" />
<meta itemprop="dateModified" content="2023-07-21T16:22:28+01:00" />
<meta itemprop="wordCount" content="1657">
<meta itemprop="keywords" content="" /><meta name="twitter:card" content="summary"/><meta name="twitter:title" content="Building trees with SKA"/>
<meta name="twitter:description" content="SKA is a tool for comparing small and highly similar genomes using split k-mers. This guide will explain how to use SKA to build a phylogenetic tree for different Escherichia coli lineages in a few minutes. Although SKA is tailored more towards analysing variation within a lineage, tree-building ends up working fine for the whole species but requires more memory.
Why SKA is good for building phylogenetic trees The basic approach to building a tree with SKA is to generate a SNP alignment using split k-mers and then feed that to a tree building algorithm of choice."/>

	



  </head>

  <body class="ma0 avenir bg-near-white">

    
   
  

  
  
  <header class="cover bg-top" style="background-image: url('http://localhost:1313/images/header1.jpg');">
    <div class="bg-black-60">
      <nav class="pv3 ph3 ph4-ns" role="navigation">
  <div class="flex-l justify-between items-center center">
    <a href="/" class="f3 fw2 hover-white no-underline white-90 dib">
      
        <img src="/images/bacpop_logo_3.svg" class="w100 mw5-ns" alt="Pathogen Informatics and Modelling Group" />
      
    </a>
    <div class="flex-l items-center">
      

      
        <ul class="pl0 mr3">
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/group/" title="About us page">
              About us
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/blog/" title="Blogs page">
              Blogs
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/guides/" title="Guides page">
              Guides
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/jobs/" title="Jobs page">
              Jobs
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/poppunk/" title="PopPUNK databases page">
              PopPUNK databases
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/publications/" title="Publications page">
              Publications
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/research/" title="Research page">
              Research
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/resources/" title="Resources page">
              Resources
            </a>
          </li>
          
          <li class="list f5 f4-ns fw4 dib pr3">
            <a class="hover-white no-underline white-90" href="/software/" title="Software page">
              Software
            </a>
          </li>
          
        </ul>
      
      
<div class="ananke-socials">
  
    <a href="https://github.com/bacpop" target="_blank" class="github ananke-social-link link-transition stackoverflow link dib z-999 pt3 pt0-l mr1" title="bacpop github link" rel="noopener" aria-label="follow on bacpop github——Opens in a new window">
      
        <span class="icon"><svg style="enable-background:new 0 0 512 512;" version="1.1" viewBox="0 0 512 512"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" >
  <path d="M256,32C132.3,32,32,134.8,32,261.7c0,101.5,64.2,187.5,153.2,217.9c11.2,2.1,15.3-5,15.3-11.1   c0-5.5-0.2-19.9-0.3-39.1c-62.3,13.9-75.5-30.8-75.5-30.8c-10.2-26.5-24.9-33.6-24.9-33.6c-20.3-14.3,1.5-14,1.5-14   c22.5,1.6,34.3,23.7,34.3,23.7c20,35.1,52.4,25,65.2,19.1c2-14.8,7.8-25,14.2-30.7c-49.7-5.8-102-25.5-102-113.5   c0-25.1,8.7-45.6,23-61.6c-2.3-5.8-10-29.2,2.2-60.8c0,0,18.8-6.2,61.6,23.5c17.9-5.1,37-7.6,56.1-7.7c19,0.1,38.2,2.6,56.1,7.7   c42.8-29.7,61.5-23.5,61.5-23.5c12.2,31.6,4.5,55,2.2,60.8c14.3,16.1,23,36.6,23,61.6c0,88.2-52.4,107.6-102.3,113.3   c8,7.1,15.2,21.1,15.2,42.5c0,30.7-0.3,55.5-0.3,63c0,6.1,4,13.3,15.4,11C415.9,449.1,480,363.1,480,261.7   C480,134.8,379.7,32,256,32z"/>
</svg>
</span>
      
<span class="new-window"><svg  height="8px"  style="enable-background:new 0 0 1000 1000;" version="1.1" viewBox="0 0 1000 1000"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" >
<path d="M598 128h298v298h-86v-152l-418 418-60-60 418-418h-152v-86zM810 810v-298h86v298c0 46-40 86-86 86h-596c-48 0-86-40-86-86v-596c0-46 38-86 86-86h298v86h-298v596h596z" style="fill-rule:evenodd;clip-rule:evenodd;"/>
</svg>
</span></a>
  
    <a href="https://youtube.com/@bacpop" target="_blank" class="youtube ananke-social-link link-transition stackoverflow link dib z-999 pt3 pt0-l mr1" title="Bacpop/ScientificProgramming YouTube channel link" rel="noopener" aria-label="follow on Bacpop/ScientificProgramming YouTube channel——Opens in a new window">
      
        <span class="icon"><svg style="enable-background:new 0 0 67 67;" version="1.1" viewBox="0 0 67 67"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><path d="M42.527,41.34c-0.278,0-0.478,0.078-0.6,0.244  c-0.121,0.156-0.18,0.424-0.18,0.796v0.896h1.543V42.38c0-0.372-0.062-0.64-0.185-0.796C42.989,41.418,42.792,41.34,42.527,41.34z   M36.509,41.309c0.234,0,0.417,0.076,0.544,0.23c0.123,0.155,0.185,0.383,0.185,0.682v4.584c0,0.286-0.053,0.487-0.153,0.611  c-0.1,0.127-0.256,0.189-0.47,0.189c-0.148,0-0.287-0.033-0.421-0.096c-0.135-0.062-0.274-0.171-0.415-0.313v-5.531  c0.119-0.122,0.239-0.213,0.36-0.271C36.26,41.335,36.383,41.309,36.509,41.309z M41.748,44.658v1.672  c0,0.468,0.057,0.792,0.17,0.974c0.118,0.181,0.313,0.269,0.592,0.269c0.289,0,0.491-0.076,0.606-0.229  c0.114-0.153,0.175-0.489,0.175-1.013v-0.405h1.795v0.456c0,0.911-0.217,1.596-0.657,2.059c-0.435,0.459-1.089,0.687-1.958,0.687  c-0.781,0-1.398-0.242-1.847-0.731c-0.448-0.486-0.676-1.157-0.676-2.014v-3.986c0-0.768,0.249-1.398,0.742-1.882  c0.493-0.484,1.128-0.727,1.911-0.727c0.799,0,1.413,0.225,1.843,0.674c0.429,0.448,0.642,1.093,0.642,1.935v2.264H41.748z   M38.623,48.495c-0.271,0.336-0.669,0.501-1.187,0.501c-0.343,0-0.646-0.062-0.912-0.192c-0.267-0.129-0.519-0.327-0.746-0.601  v0.681h-1.764V36.852h1.764v3.875c0.237-0.27,0.485-0.478,0.748-0.616c0.267-0.143,0.534-0.212,0.805-0.212  c0.554,0,0.975,0.189,1.265,0.565c0.294,0.379,0.438,0.933,0.438,1.66v4.926C39.034,47.678,38.897,48.159,38.623,48.495z   M30.958,48.884v-0.976c-0.325,0.361-0.658,0.636-1.009,0.822c-0.349,0.191-0.686,0.282-1.014,0.282  c-0.405,0-0.705-0.129-0.913-0.396c-0.201-0.266-0.305-0.658-0.305-1.189v-7.422h1.744v6.809c0,0.211,0.037,0.362,0.107,0.457  c0.077,0.095,0.196,0.141,0.358,0.141c0.128,0,0.292-0.062,0.488-0.188c0.197-0.125,0.375-0.283,0.542-0.475v-6.744h1.744v8.878  H30.958z M24.916,38.6v10.284h-1.968V38.6h-2.034v-1.748h6.036V38.6H24.916z M32.994,32.978c0-0.001,12.08,0.018,13.514,1.45  c1.439,1.435,1.455,8.514,1.455,8.555c0,0-0.012,7.117-1.455,8.556C45.074,52.969,32.994,53,32.994,53s-12.079-0.031-13.516-1.462  c-1.438-1.435-1.441-8.502-1.441-8.556c0-0.041,0.004-7.12,1.441-8.555C20.916,32.996,32.994,32.977,32.994,32.978z M42.52,29.255  h-1.966v-1.08c-0.358,0.397-0.736,0.703-1.13,0.909c-0.392,0.208-0.771,0.312-1.14,0.312c-0.458,0-0.797-0.146-1.027-0.437  c-0.229-0.291-0.345-0.727-0.345-1.311v-8.172h1.962v7.497c0,0.231,0.045,0.399,0.127,0.502c0.08,0.104,0.216,0.156,0.399,0.156  c0.143,0,0.327-0.069,0.548-0.206c0.22-0.137,0.423-0.312,0.605-0.527v-7.422h1.966V29.255z M31.847,27.588  c0.139,0.147,0.339,0.219,0.6,0.219c0.266,0,0.476-0.075,0.634-0.223c0.157-0.152,0.235-0.358,0.235-0.618v-5.327  c0-0.214-0.08-0.387-0.241-0.519c-0.16-0.131-0.37-0.196-0.628-0.196c-0.241,0-0.435,0.065-0.586,0.196  c-0.148,0.132-0.225,0.305-0.225,0.519v5.327C31.636,27.233,31.708,27.439,31.847,27.588z M30.408,19.903  c0.528-0.449,1.241-0.674,2.132-0.674c0.812,0,1.48,0.237,2.001,0.711c0.517,0.473,0.777,1.083,0.777,1.828v5.051  c0,0.836-0.255,1.491-0.762,1.968c-0.513,0.476-1.212,0.714-2.106,0.714c-0.858,0-1.547-0.246-2.064-0.736  c-0.513-0.492-0.772-1.152-0.772-1.983v-5.068C29.613,20.954,29.877,20.351,30.408,19.903z M24.262,16h-2.229l2.634,8.003v5.252  h2.213v-5.5L29.454,16h-2.25l-1.366,5.298h-0.139L24.262,16z M33,64C16.432,64,3,50.569,3,34S16.432,4,33,4s30,13.431,30,30  S49.568,64,33,64z" style="fill-rule:evenodd;clip-rule:evenodd;"/></svg>
</span>
      
<span class="new-window"><svg  height="8px"  style="enable-background:new 0 0 1000 1000;" version="1.1" viewBox="0 0 1000 1000"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" >
<path d="M598 128h298v298h-86v-152l-418 418-60-60 418-418h-152v-86zM810 810v-298h86v298c0 46-40 86-86 86h-596c-48 0-86-40-86-86v-596c0-46 38-86 86-86h298v86h-298v596h596z" style="fill-rule:evenodd;clip-rule:evenodd;"/>
</svg>
</span></a>
  
</div>

    </div>
  </div>
</nav>

      <div class="tc-l pv6 ph3 ph4-ns">
        
          <h1 class="f2 f1-l fw2 white-90 mb0 lh-title">Building trees with SKA</h1>
          
        
      </div>
    </div>
  </header>



    <main class="pb7" role="main">
      
  
  <article class="flex-l flex-wrap justify-between mw8 center ph3">
    <header class="mt4 w-100">
      <aside class="instapaper_ignoref b helvetica tracked">
          
        GUIDES
      </aside>
      










  <div id="sharing" class="mt3 ananke-socials">
    
  </div>


      <h1 class="f1 athelas mt3 mb1">Building trees with SKA</h1>
      
      <p class="tracked">
        By <strong>Tommi Mäklin</strong>
      </p>
      
      
      
      <time class="f6 mv4 dib tracked" datetime="2023-07-21T16:22:28+01:00">July 21, 2023</time>
      

      
      
    </header>
    <div class="nested-copy-line-height lh-copy serif f4 nested-links mid-gray pr4-l w-two-thirds-l"><p><a href="https://github.com/bacpop/ska.rust">SKA</a> is a tool for comparing small and highly similar genomes using <a href="https://www.biorxiv.org/content/early/2018/10/25/453142">split <em>k</em>-mers</a>. This guide will explain how to use SKA to build a phylogenetic tree for different <em>Escherichia coli</em> lineages in a few minutes. Although SKA is tailored more towards analysing variation within a lineage, tree-building ends up working fine for the whole species but requires more memory.</p>
<p><img src="/images/ska-trees/ska_logo.png" alt="SKA logo"></p>
<h2 id="why-ska-is-good-for-building-phylogenetic-trees">Why SKA is good for building phylogenetic trees</h2>
<p>The basic approach to building a tree with SKA is to generate a SNP alignment using split <em>k</em>-mers and then feed that to a tree building algorithm of choice. Since SKA does not require a specifying a reference sequence to determine the SNPs, SKA gets around potential biases introduced by reference choice and is thus particularly well suited to analysing microbial genomes derived from outbreaks or pathogen surveillance.</p>
<h2 id="installing-ska">Installing SKA</h2>
<p>SKA can be installed in multiple ways:</p>
<ol>
<li>Precompiled binary from the <a href="https://github.com/bacpop/ska.rust/releases">SKA GitHub repository</a>.</li>
<li>Through the rust programming language&rsquo;s package manager cargo with <code>cargo install ska</code>.</li>
<li>From bioconda with <code>conda install -c bioconda ska2</code>.</li>
<li>Building from source.</li>
</ol>
<p>In this tutorial I used SKA v0.3.2 to run everything.</p>
<h2 id="input-data">Input data</h2>
<p>We will use a collection of 46 <em>E. coli</em> hybrid Nanopore+Illumina assemblies isolated from travellers visiting the city of Vientiane in Laos. These assemblies were published as part of <a href="https://doi.org/10.1099/mgen.0.001000">Snaith et al. (2023)</a>.</p>
<p>All materials for running the analysis in this post can be downloaded from <a href="https://zenodo.org/record/8172518/files/building_trees_with_ska.tar">zenodo</a> with wget</p>
<pre tabindex="0"><code>wget https://zenodo.org/record/8172518/files/building_trees_with_ska.tar
</code></pre><p>After downloading the assemblies, run <code>tar -xvf building_trees_with_ska.tar</code> to extract the files.</p>
<h2 id="reference-free-workflow">Reference-free workflow</h2>
<p>The first example describes using a reference-free workflow to analyse assemblies in a case where we are only interested in the SNPs and do not care about structural rearrangements or other global changes. This is accomplished with the <code>ska align</code> command and the result can be used to e. g. build a tree for across-species variation.</p>
<h3 id="indexing-the-assemblies">Indexing the assemblies</h3>
<p>To run the split <em>k</em>-mer alignment, we first need to build an index of the split <em>k</em>-mers from the assemblies. This is accomplished with the <code>ska build</code> command which requires a tab-separated list consisting of the sequence name and assembly file path as input:</p>
<pre tabindex="0"><code>LA002	assemblies/LA002.fa.gz
LA022	assemblies/LA022.fa.gz
LA023	assemblies/LA023.fa.gz
.
.
.
</code></pre><p>which is provided in <code>data/laos_ska_input.tsv</code>, or can be created using the command</p>
<pre tabindex="0"><code>paste &lt;(ls assemblies | sed &#39;s/[.].*$//g&#39;) &lt;(ls -d assemblies/*) &gt; data/laos_ska_input.tsv
</code></pre><p>The above only works if you&rsquo;re using bash as your shell.</p>
<p>The split <em>k</em>-mers index is built from the input list by running</p>
<pre tabindex="0"><code>ska build -f data/laos_ska_input.tsv -k 31 -o output/laos_ska_index --threads 4
</code></pre><p>which will take a few minutes and use ~8GB of memory. SKA uses a lot of memory for species-wide analyses but when doing within-lineage analyses the method would be more efficient.</p>
<p><img src="/images/ska-trees/ska_run.png" alt="Screenshot of the output written by SKA to terminal"></p>
<p>In the example we chose the value of <em>k</em> as 31 which is a commonly used choice for analysing bacterial genomes. For a more detailed analysis of the different possible values of <em>k</em>, have a look at the paper by <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0258693">Bussi, Kapon, and Reich</a> in PLOS ONE.</p>
<h3 id="producing-the-snp-alignment">Producing the SNP alignment</h3>
<p>The SNP alignment can be extracted from the .skf file produced in the previous step by running</p>
<pre tabindex="0"><code>ska align --min-freq 1 --filter no-filter output/laos_ska_index.skf -o output/laos_ska_alignment.aln --threads 4
</code></pre><p>which will again take around a few minutes and use around 1.5GB memory. The resulting alignment has around 435k sites.</p>
<p>In the above command, the <code>--min-freq</code> option sets the maximum number of missing sites in the alignment to 1 and the <code>--filter no-filter</code> option specifies that no extra filtering should be performed. Other options include filtering all constant sites with <code>--filter no-const</code> or all constant and ambiguous sites with <code>--filter no-ambig-or-const</code>. Using the last option is equivalent to treating any ambiguous site as an N.</p>
<h3 id="build-a-tree">Build a tree</h3>
<p>If no further pruning is needed, we can just use the alignment from the previous step to build a phylogeny. A good tool for doing this quickly is <a href="https://github.com/citiususc/veryfasttree">VeryFastTree</a> but if you have a lot of time <a href="https://github.com/amkozlov/raxml-ng">RAxML</a> is also a good choice. Both can be installed from bioconda.</p>
<p>Build the tree with VeryFastTree (v4.0.1) by running</p>
<pre tabindex="0"><code>VeryFastTree -nt -gamma -gtr -threads 4 output/laos_ska_alignment.aln &gt; output/laos_ska_tree.tree
</code></pre><p>As the name might suggest, VeryFastTree is fast and manages to build the tree in a few minutes.</p>
<h3 id="look-at-the-tree">Look at the tree</h3>
<p>We can use the minimalistic <a href="https://github.com/iBiology/plottree">plottree</a> (install with <code>pip install plottree</code>) to quickly look at the tree using</p>
<pre tabindex="0"><code>plottree output/laos_ska_tree.tree
</code></pre><p>This shows that some of the assemblies are more closely related than the others.</p>
<p><img src="/images/ska-trees/plottree_ska.png" alt="Phylogenetic tree from raw SKA output"></p>
<h3 id="add-in-some-colours">Add in some colours</h3>
<p>To build a fancier tree, we can use the R script provided at the end of this post to visualize and <a href="https://www.ebi.ac.uk/training/online/courses/introduction-to-phylogenetics/what-is-a-phylogeny/aspects-of-phylogenies/root/">midpoint root</a> the tree with the leaves coloured according to PopPUNK clusters provided in <code>data/laos_poppunk_clusters.tsv</code>. Adding the cluster colours and labels shows that SKA + VeryFastTree has managed to infer a phylogeny that preserves their locality (defined as two assemblies belonging to the same cluster).</p>
<p><img src="/images/ska-trees/SKA_tree_fancy.png" alt="A phylogenetic trees constructed with SKA + VeryFastTree and coloured by PopPUNK cluster. The topography nicely corresponds with the clusters."></p>
<p>Next, we&rsquo;ll take a look at using the other mode SKA provides for producing alignments: mapping against a reference sequence.</p>
<h2 id="alignment-based-workflow">Alignment-based workflow</h2>
<p>In addition to the reference-free alignments, SKA can also be used to produce alignments against a reference genome. This is useful if the order of the output alignment is meaningful, such as when planning to use <a href="https://github.com/nickjcroucher/gubbins">gubbins</a> to remove recombination.</p>
<p>Note that gubbins is not suitable for analysing variation across a species (see the note about <em>&ldquo;What type of dataset can be analysed with Gubbins?&rdquo;</em> in the <a href="https://nickjcroucher.github.io/gubbins/">gubbins documentation</a> for more details). To avoid running gubbins on unsuitable data, we&rsquo;ll focus our analysis on variation within the largest PopPUNK cluster in the collection, SC4. This cluster corresponds to the <em>E. coli</em> ST10 clonal complex.</p>
<h3 id="reference-sequence-for-sc4">Reference sequence for SC4</h3>
<p>Fortunately for us, ST10 has a downloadable a reference genome in the ENA that we can download with wget</p>
<pre tabindex="0"><code>wget -q -O - https://www.ebi.ac.uk/ena/browser/api/fasta/U00096.3?download=true | gzip &gt; GCA_000005845.2.fna.gz
</code></pre><p>Since we now only want to use a subset of the assemblies, we&rsquo;ll need to rebuild the index for the genomes that belong to SC4. This info is contained in the <code>data/laos_poppunk_clusters.tsv</code> file and we use the following commands to both convert it to an input list for SKA and to add the reference sequence to the index</p>
<pre tabindex="0"><code>paste &lt;(grep &#34;SC4$&#34; data/laos_poppunk_clusters.tsv | cut -f1) &lt;(grep &#34;SC4$&#34; data/laos_poppunk_clusters.tsv | cut -f1 | sed &#39;s/^/assemblies\//g&#39; | sed &#39;s/$/.fa.gz/g&#39;) &gt; output/laos_SC4_input_list.tsv
echo -e &#34;GCA_000005845.2\t$(pwd)/GCA_000005845.2.fna.gz&#34; &gt;&gt; output/laos_SC4_input_list.tsv
</code></pre><h3 id="aligning-with-ska-map">Aligning with ska map</h3>
<p>Build the index as previously with <code>ska build</code> and then run <code>ska map</code> to align the indexed assemblies against the reference genome with</p>
<pre tabindex="0"><code>ska build -f output/laos_SC4_input_list.tsv -k 31 -o output/laos_SC4_index --threads 4
ska map -o output/laos_SC4_map.aln --ambig-mask --threads 4 GCA_000005845.2.fna.gz output/laos_SC4_index.skf
</code></pre><p>This time the alignment has 4 641 653 sites, which is the number of bases in the reference genome. Compared to <code>ska align</code>, using <code>ska map</code> keeps all bases in the reference sequence and replaces the sites it cannot find in the indexed assemblies with gaps. If you use the command</p>
<pre tabindex="0"><code>sed -n &#39;4p&#39; output/laos_SC4_map.aln | less
</code></pre><p>to look at the alignment of the first assembly against the reference you&rsquo;ll see that some sites are marked as missing/gaps.</p>
<h3 id="removing-recombination">Removing recombination</h3>
<p>In a typical genomic epidemiology workflow looking at variants <strong>within a lineage</strong> one typically wants to remove recombination from the alignment. For our SC4 alignment, we will do this using <a href="https://github.com/nickjcroucher/gubbins">gubbins</a>. Since Gubbins only accepts characters in the ACGTN- range, we had to use the <code>--ambig-mask</code> option with <code>ska map</code> to mask ambiguous bases (SKA uses the <a href="https://en.wikipedia.org/wiki/Nucleic_acid_notation">IUPAC notation</a> which gubbins does not support) with N&rsquo;s.</p>
<p>Gubbins (v3.2.1 in our example) can be run on the reference mapping with</p>
<pre tabindex="0"><code>run_gubbins.py --prefix output/gubbins output/laos_SC4_map.aln --threads 4 --filter-percentage 27.0
</code></pre><p>This will create the recombination-free tree <code>output/gubbins.final_tree.tre</code>. We had to increase the <code>--filter-percentage</code> parameter from the default 25.0 to 27.0 because one of the input sequences had ~26% missing sites.</p>
<p>Gubbins produces an alignment where recombination has been removed and only ~7 000 sites remain. These sites are enough to differentiate the assemblies belonging to SC4 and we can check this by visualising the tree with plottree</p>
<pre tabindex="0"><code>plottree output/gubbins.final_tree.tre
</code></pre><p><img src="/images/ska-trees/plottree_sc4.png" alt="Phylogenetic tree from Gubbins"></p>
<p>In this tree the assemblies from Laos cluster into several distinct groups within the lineage, and none of them are closely related to the reference genome which is sensible since the reference genome is from the <em>E. coli</em> K-12 strain that was isolated a hundred years ago on the other side of the world in <a href="https://www.genome.wisc.edu/resources/strains.htm">Palo Alto, California, USA in 1922</a>.</p>
<p>And that&rsquo;s it! Happy tree-building.</p>
<h2 id="about-the-author">About the author</h2>
<p>I visited John Lees&rsquo; lab in Spring 2023 and have been using both SKA v1 and v2 to build lots of trees. This guide was a result of wanting to get others to do the same because installing snippy gets increasingly more difficult over time and reference-free approaches should be preferred anyway. If you want to know what I&rsquo;m doing now, check out my personal website <a href="https://maklin.fi">https://maklin.fi</a>.</p>
<h2 id="r-script-for-plotting-trees">R script for plotting trees</h2>
<pre tabindex="0"><code>#!/usr/bin/env Rscript
#
## Plot a tree and add some colors

library(&#34;ape&#34;)
library(&#34;phytools&#34;)

## Okabe-Ito color palette for the 7 most common clusters
colors &lt;- c(&#34;#e69f00&#34;, &#34;#56b4e9&#34;, &#34;#009e73&#34;, &#34;#f0e442&#34;, &#34;#0072b2&#34;, &#34;#d55e00&#34;, &#34;#cc79a7&#34;)

## Read the raw SKA alignment + VeryFastTree tree in and midpoint root it
tree2 &lt;- read.tree(&#34;ska.aln.tree&#34;)
midpoint_rooted_tree &lt;- midpoint.root(tree2)

## Read in the PopPUNK clusters for each assembly
sts &lt;- read.table(&#34;data/laos_poppunk_clusters.tsv&#34;, sep=&#39;\t&#39;, comment.char=&#39;@&#39;, header=FALSE)
sts &lt;- sts[match(tree2$tip.label, sts$V1), ]

## Get the 7 most common clusters
top_st_names &lt;- names(tail(sort(table(sts$V2)), 7))
top_st_colors &lt;- cbind(top_st_names, colors[as.numeric(factor(top_st_names))])

## Set up the colors so that the 7 most common have a different color from the rest
other_st_names &lt;- setdiff(unique(sts$V2), top_st_names)
st_colors &lt;- rbind(top_st_colors, cbind(other_st_names, rep(&#34;black&#34;, length(other_st_names))))
st_colors &lt;- st_colors[match(sts[, 2], st_colors), ]

## Add colors and labels, same as above
st_colors &lt;- st_colors[match(sts[, 2], st_colors), ]
tree2$tip.label &lt;- paste(tree2$tip.label, sts[, 2], sep=&#39;_&#39;)
tree2.colors &lt;- st_colors[, 2]

## Plot both trees side-by-side
pdf(file=&#34;SKA_tree_fancy.pdf&#34;, width=9, height=9)
par(mar=c(1, 1, 3, 1))
plot.phylo(tree2, tip.color=tree2.colors)
title(main=&#34;Maximum likelihood tree\n(SKA + VeryFastTree)&#34;, adj=0, font.main=1, cex.main=1.5)
legend(&#34;bottomright&#34;, legend=top_st_colors[order(as.numeric(gsub(&#34;SC&#34;, &#34;&#34;, top_st_colors[, 1]))), 1], fill=top_st_colors[order(as.numeric(gsub(&#34;SC&#34;, &#34;&#34;, top_st_colors[, 1]))), 2], bty=&#39;n&#39;, title=&#34;PopPUNK\ncluster&#34;, cex=1.33)
dev.off()
</code></pre><ul class="pa0">
  
</ul>
<div class="mt6 instapaper_ignoref">
      
      
      </div>
    </div>

    <aside class="w-30-l mt6-l">




</aside>

  </article>

    </main>
    <footer class="bg-black bottom-0 w-100 pa3" role="contentinfo">
  <div class="flex justify-between">
  <a class="f4 fw4 hover-white no-underline white-70 dn dib-ns pv2 ph3" href="http://localhost:1313/" >
    &copy;  Pathogen Informatics and Modelling Group 2024 
  </a>
    <div>
<div class="ananke-socials">
  
    <a href="https://github.com/bacpop" target="_blank" class="github ananke-social-link link-transition stackoverflow link dib z-999 pt3 pt0-l mr1" title="bacpop github link" rel="noopener" aria-label="follow on bacpop github——Opens in a new window">
      
        <span class="icon"><svg style="enable-background:new 0 0 512 512;" version="1.1" viewBox="0 0 512 512"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" >
  <path d="M256,32C132.3,32,32,134.8,32,261.7c0,101.5,64.2,187.5,153.2,217.9c11.2,2.1,15.3-5,15.3-11.1   c0-5.5-0.2-19.9-0.3-39.1c-62.3,13.9-75.5-30.8-75.5-30.8c-10.2-26.5-24.9-33.6-24.9-33.6c-20.3-14.3,1.5-14,1.5-14   c22.5,1.6,34.3,23.7,34.3,23.7c20,35.1,52.4,25,65.2,19.1c2-14.8,7.8-25,14.2-30.7c-49.7-5.8-102-25.5-102-113.5   c0-25.1,8.7-45.6,23-61.6c-2.3-5.8-10-29.2,2.2-60.8c0,0,18.8-6.2,61.6,23.5c17.9-5.1,37-7.6,56.1-7.7c19,0.1,38.2,2.6,56.1,7.7   c42.8-29.7,61.5-23.5,61.5-23.5c12.2,31.6,4.5,55,2.2,60.8c14.3,16.1,23,36.6,23,61.6c0,88.2-52.4,107.6-102.3,113.3   c8,7.1,15.2,21.1,15.2,42.5c0,30.7-0.3,55.5-0.3,63c0,6.1,4,13.3,15.4,11C415.9,449.1,480,363.1,480,261.7   C480,134.8,379.7,32,256,32z"/>
</svg>
</span>
      
<span class="new-window"><svg  height="8px"  style="enable-background:new 0 0 1000 1000;" version="1.1" viewBox="0 0 1000 1000"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" >
<path d="M598 128h298v298h-86v-152l-418 418-60-60 418-418h-152v-86zM810 810v-298h86v298c0 46-40 86-86 86h-596c-48 0-86-40-86-86v-596c0-46 38-86 86-86h298v86h-298v596h596z" style="fill-rule:evenodd;clip-rule:evenodd;"/>
</svg>
</span></a>
  
    <a href="https://youtube.com/@bacpop" target="_blank" class="youtube ananke-social-link link-transition stackoverflow link dib z-999 pt3 pt0-l mr1" title="Bacpop/ScientificProgramming YouTube channel link" rel="noopener" aria-label="follow on Bacpop/ScientificProgramming YouTube channel——Opens in a new window">
      
        <span class="icon"><svg style="enable-background:new 0 0 67 67;" version="1.1" viewBox="0 0 67 67"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><path d="M42.527,41.34c-0.278,0-0.478,0.078-0.6,0.244  c-0.121,0.156-0.18,0.424-0.18,0.796v0.896h1.543V42.38c0-0.372-0.062-0.64-0.185-0.796C42.989,41.418,42.792,41.34,42.527,41.34z   M36.509,41.309c0.234,0,0.417,0.076,0.544,0.23c0.123,0.155,0.185,0.383,0.185,0.682v4.584c0,0.286-0.053,0.487-0.153,0.611  c-0.1,0.127-0.256,0.189-0.47,0.189c-0.148,0-0.287-0.033-0.421-0.096c-0.135-0.062-0.274-0.171-0.415-0.313v-5.531  c0.119-0.122,0.239-0.213,0.36-0.271C36.26,41.335,36.383,41.309,36.509,41.309z M41.748,44.658v1.672  c0,0.468,0.057,0.792,0.17,0.974c0.118,0.181,0.313,0.269,0.592,0.269c0.289,0,0.491-0.076,0.606-0.229  c0.114-0.153,0.175-0.489,0.175-1.013v-0.405h1.795v0.456c0,0.911-0.217,1.596-0.657,2.059c-0.435,0.459-1.089,0.687-1.958,0.687  c-0.781,0-1.398-0.242-1.847-0.731c-0.448-0.486-0.676-1.157-0.676-2.014v-3.986c0-0.768,0.249-1.398,0.742-1.882  c0.493-0.484,1.128-0.727,1.911-0.727c0.799,0,1.413,0.225,1.843,0.674c0.429,0.448,0.642,1.093,0.642,1.935v2.264H41.748z   M38.623,48.495c-0.271,0.336-0.669,0.501-1.187,0.501c-0.343,0-0.646-0.062-0.912-0.192c-0.267-0.129-0.519-0.327-0.746-0.601  v0.681h-1.764V36.852h1.764v3.875c0.237-0.27,0.485-0.478,0.748-0.616c0.267-0.143,0.534-0.212,0.805-0.212  c0.554,0,0.975,0.189,1.265,0.565c0.294,0.379,0.438,0.933,0.438,1.66v4.926C39.034,47.678,38.897,48.159,38.623,48.495z   M30.958,48.884v-0.976c-0.325,0.361-0.658,0.636-1.009,0.822c-0.349,0.191-0.686,0.282-1.014,0.282  c-0.405,0-0.705-0.129-0.913-0.396c-0.201-0.266-0.305-0.658-0.305-1.189v-7.422h1.744v6.809c0,0.211,0.037,0.362,0.107,0.457  c0.077,0.095,0.196,0.141,0.358,0.141c0.128,0,0.292-0.062,0.488-0.188c0.197-0.125,0.375-0.283,0.542-0.475v-6.744h1.744v8.878  H30.958z M24.916,38.6v10.284h-1.968V38.6h-2.034v-1.748h6.036V38.6H24.916z M32.994,32.978c0-0.001,12.08,0.018,13.514,1.45  c1.439,1.435,1.455,8.514,1.455,8.555c0,0-0.012,7.117-1.455,8.556C45.074,52.969,32.994,53,32.994,53s-12.079-0.031-13.516-1.462  c-1.438-1.435-1.441-8.502-1.441-8.556c0-0.041,0.004-7.12,1.441-8.555C20.916,32.996,32.994,32.977,32.994,32.978z M42.52,29.255  h-1.966v-1.08c-0.358,0.397-0.736,0.703-1.13,0.909c-0.392,0.208-0.771,0.312-1.14,0.312c-0.458,0-0.797-0.146-1.027-0.437  c-0.229-0.291-0.345-0.727-0.345-1.311v-8.172h1.962v7.497c0,0.231,0.045,0.399,0.127,0.502c0.08,0.104,0.216,0.156,0.399,0.156  c0.143,0,0.327-0.069,0.548-0.206c0.22-0.137,0.423-0.312,0.605-0.527v-7.422h1.966V29.255z M31.847,27.588  c0.139,0.147,0.339,0.219,0.6,0.219c0.266,0,0.476-0.075,0.634-0.223c0.157-0.152,0.235-0.358,0.235-0.618v-5.327  c0-0.214-0.08-0.387-0.241-0.519c-0.16-0.131-0.37-0.196-0.628-0.196c-0.241,0-0.435,0.065-0.586,0.196  c-0.148,0.132-0.225,0.305-0.225,0.519v5.327C31.636,27.233,31.708,27.439,31.847,27.588z M30.408,19.903  c0.528-0.449,1.241-0.674,2.132-0.674c0.812,0,1.48,0.237,2.001,0.711c0.517,0.473,0.777,1.083,0.777,1.828v5.051  c0,0.836-0.255,1.491-0.762,1.968c-0.513,0.476-1.212,0.714-2.106,0.714c-0.858,0-1.547-0.246-2.064-0.736  c-0.513-0.492-0.772-1.152-0.772-1.983v-5.068C29.613,20.954,29.877,20.351,30.408,19.903z M24.262,16h-2.229l2.634,8.003v5.252  h2.213v-5.5L29.454,16h-2.25l-1.366,5.298h-0.139L24.262,16z M33,64C16.432,64,3,50.569,3,34S16.432,4,33,4s30,13.431,30,30  S49.568,64,33,64z" style="fill-rule:evenodd;clip-rule:evenodd;"/></svg>
</span>
      
<span class="new-window"><svg  height="8px"  style="enable-background:new 0 0 1000 1000;" version="1.1" viewBox="0 0 1000 1000"  xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" >
<path d="M598 128h298v298h-86v-152l-418 418-60-60 418-418h-152v-86zM810 810v-298h86v298c0 46-40 86-86 86h-596c-48 0-86-40-86-86v-596c0-46 38-86 86-86h298v86h-298v596h596z" style="fill-rule:evenodd;clip-rule:evenodd;"/>
</svg>
</span></a>
  
</div>
</div>
  </div>
</footer>

  </body>
</html>
